### PR TITLE
#165413068 add functionalities for creating and updating an article

### PR DESCRIPTION
**Pull request by Niyitangasam**
Description
=======

This PR is for adding functionalities of  creating and updating an article

Type of change
=======
N/A

How has it been tested
=======
It can be tested by log in and access below  routes:

- For creating a new article: ``/create``
- For updating an existing article: ``/update/slug`` but you must be the author.

Checklist:
=======
- Add components for creating and updating the article.
- Add its actions and reducers
- Add necessary HTML and scss to the page
- Add tests.

Pivotal tracker story ID
=======
[#165413068](https://www.pivotaltracker.com/n/projects/2326984/stories/165413068)

Screenshots
=======
image

image

### CODE REVIEW

File: src/setupTests.js [hunk 1]
```diff
@@ -8,3 +8,10 @@ global.React = React;
 global.shallow = shallow;
 global.render = render;
 global.mount = mount;
+global.MutationObserver = class {
+  constructor() {}
+  disconnect() {}
+  observe() {}
+  takeRecords() {return []}
+};
+global.document.getSelection = function() {}
```
Comment (AJAkimana) [inline src/setupTests.js]: Add a new line at the end of this file

File: src/setupTests.js [hunk 2]
```diff
@@ -8,3 +8,10 @@ global.React = React;
 global.shallow = shallow;
 global.render = render;
 global.mount = mount;
+global.MutationObserver = class {
```
Comment (AJAkimana) [inline src/setupTests.js:11]: What these lines of code do? Just try to explain

**Review by AJAkimana — COMMENTED** [review]
Good work @Niyitangasam, Try to take a look at the inline comments

File: src/setupTests.js (hunk 2 above)
Comment (Niyitangasam) [reply src/setupTests.js:11]: While I was testing, I realized that React quill needs mutation observer in its testing. Mutation observer helps in tracking changes while typing.

You can refer to this issue  https://github.com/zenoamaro/react-quill/issues/434

**Review by AJAkimana — APPROVED** [review]
Good work @Niyitangasam

**Review by jabichris — APPROVED** [review]
Good work @Niyitangasam

**Review by AJAkimana — APPROVED** [review]
Great work

**Pull request merged**